utils/s2n_mem.c

```diff
@@ -153,6 +153,7 @@ int s2n_alloc(struct s2n_blob *b, uint32_t size)
 {
     S2N_ERROR_IF(initialized == false, S2N_ERR_NOT_INITIALIZED);
     notnull_check(b);
+    POSIX_ENSURE(b->size == 0, S2N_ERR_ALLOC);
```

Comment:
Shouldn't you be checking b->allocated instead of b->size?

Reply:
Ah right. I'll change it